odxtools/createanydiagcodedtype.py

```diff
@@ -1,85 +1,28 @@
-        return LeadingLengthInfoType(
-            base_data_type=base_data_type,
-            bit_length=bit_length,
-            base_type_encoding=base_type_encoding,
-            is_highlow_byte_order_raw=is_highlow_byte_order_raw,
-        )
+        return LeadingLengthInfoType.from_et(et_element, doc_frags)
     elif dct_type == "MIN-MAX-LENGTH-TYPE":
-        min_length = int(odxrequire(et_element.findtext("MIN-LENGTH")))
-        max_length = None
-        if et_element.find("MAX-LENGTH") is not None:
-            max_length = int(odxrequire(et_element.findtext("MAX-LENGTH")))
-        termination = odxrequire(et_element.get("TERMINATION"))
-
-        return MinMaxLengthType(
-            base_data_type=base_data_type,
-            min_length=min_length,
-            max_length=max_length,
-            termination=termination,
-            base_type_encoding=base_type_encoding,
-            is_highlow_byte_order_raw=is_highlow_byte_order_raw,
-        )
+        return MinMaxLengthType.from_et(et_element, doc_frags)
     elif dct_type == "PARAM-LENGTH-INFO-TYPE":
-        length_key_ref = odxrequire(
-            OdxLinkRef.from_et(et_element.find("LENGTH-KEY-REF"), doc_frags))
-
-        return ParamLengthInfoType(
-            base_data_type=base_data_type,
-            length_key_ref=length_key_ref,
-            base_type_encoding=base_type_encoding,
-            is_highlow_byte_order_raw=is_highlow_byte_order_raw,
-        )
+        return ParamLengthInfoType.from_et(et_element, doc_frags)
     elif dct_type == "STANDARD-LENGTH-TYPE":
-        bit_length = int(odxrequire(et_element.findtext("BIT-LENGTH")))
-        bit_mask = None
-        if (bit_mask_str := et_element.findtext("BIT-MASK")) is not None:
-            # The XSD uses the type xsd:hexBinary
-            # xsd:hexBinary allows for leading/trailing whitespace, empty strings, and it only allows an even
-            # number of hex digits, while some of the examples shown in the  ODX specification exhibit an
-            # odd number of hex digits.
-            # This causes a validation paradox, so we try to be flexible
-            bit_mask_str = bit_mask_str.strip()
-            if len(bit_mask_str):
-                bit_mask = int(bit_mask_str, 16)
-        is_condensed_raw = odxstr_to_bool(et_element.get("CONDENSED"))
-        return StandardLengthType(
-            base_data_type=base_data_type,
-            bit_length=bit_length,
-            bit_mask=bit_mask,
-            is_condensed_raw=is_condensed_raw,
-            base_type_encoding=base_type_encoding,
-            is_highlow_byte_order_raw=is_highlow_byte_order_raw,
-        )
-    raise NotImplementedError(f"I do not know the diag-coded-type {dct_type}")
+        return StandardLengthType.from_et(et_element, doc_frags)
+
+    odxraise(f"Unknown DIAG-CODED-TYPE {dct_type}", NotImplementedError)
+    return DiagCodedType.from_et(et_element, doc_frags)
```

Comment:
`NotSupportedDtc.from_et` will only be used here instead of the abstract class

Reply:
yes, but I think it is weird to derive from one class and call the constructor of another one which is seemingly unrelated and has a very confusing name. Also, this would add a basically empty class to the code base. tl;dr: Although far from perfect, I vastly prefer the "manual ABC" option...